### PR TITLE
Make websockets more friendly to the client

### PR DESCRIPTION
**Pull request by kestivvi**
Now server and client are much stronger linked. Client directly uses classes from the server through jitpack and because of that, there was need to refactor a little bit classes and their names, so the client wouldn't be so confused.

Additionally now messages are objects, no longer sending jsons around (but you can if you like, but objects require you to conform to some standards and they are type checked, so they are more secure)

Now if server reaction wants some data provided by the user, it have to include `public static` nested class annotated with `@DataClass`. `static` is needed so that client can instantiate those classes. Not only data class has to be `static` and `public`, composite classes as well.

Each server reaction has to have response dataclass as well, so that client knows exactly what they can receive and parse it ridiculously easily.

Also other minor changes like using UUID type and enums type instead of strings to define user request type and server message/response type.

### CODE REVIEW

**Comment by kestivvi** [comment]
See related [pull request](https://github.com/Splendor-Mobile-Game/ClientSide/pull/16) on the client side. Those two requests must be merged simultaneously to work.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java [hunk 1]
```diff
@@ -80,24 +142,12 @@ public void react() {
...
-            response.addProperty("type", ResponseType.CREATE_ROOM_RESPONSE.toString());
-            response.addProperty("result", Result.OK.toString());
-            response.add("data", data);
+            UserDataResponse userDataResponse = new UserDataResponse(dataDTO.userDTO.uuid, dataDTO.userDTO.name);
+            RoomDataResponse roomDataResponse = new RoomDataResponse(dataDTO.roomDTO.name);
+            ResponseData responseData = new ResponseData(userDataResponse, roomDataResponse);
+            Response response = new Response(UUID.fromString(receivedMessage.getMessageContextId()), ResponseType.CREATE_ROOM_RESPONSE.toString(), Result.OK, responseData);
 
-            messenger.addMessageToSend(this.connectionHashCode, (new Gson()).toJson(response));
+            messenger.addMessageToSend(this.connectionHashCode, response);
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java]: `"Additionally now messages are objects, no longer sending jsons around"`. Finally it is still converted to json. What you're saying in pull request description isn't true

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/JoinRoom.java [hunk 2]
```diff
@@ -78,28 +112,14 @@ public void react() {
...
-
+            RoomDataResponse roomData = new RoomDataResponse(room.getUuid(), room.getName());
+            UserDataResponse userData = new UserDataResponse(dataDTO.userDTO.uuid, user.getName());
+            ResponseData responseData = new ResponseData(userData, roomData);
+            ServerMessage responseMessage = new ServerMessage(receivedMessage.getMessageContextId(), ServerMessageType.JOIN_ROOM_RESPONSE, Result.OK, responseData);
             
             // Send join information to other players
             for (User u : room.getAllUsers()) {
-                messenger.addMessageToSend(u.getConnectionHasCode(), (new Gson()).toJson(response));
+                messenger.addMessageToSend(u.getConnectionHasCode(), (new Gson()).toJson(responseMessage));
```
Comment (Vertonowsky) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/JoinRoom.java]: Once you pass `response` variable, object of `ServerMessage` type. And next time you use `(new Gson()).toJson(responseMessage)`. Decide which version you want to apply to the whole code

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java (hunk 1 above)
Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java]: True, I wasn't consistent. I've fixed it now.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/JoinRoom.java (hunk 2 above)
Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/JoinRoom.java]: Fixed.

**Review by Vertonowsky — DISMISSED** [review]

**Review by Vertonowsky — APPROVED** [review]

**Pull request merged**